main/src/cgeo/geocaching/utils/Formatter.java

```diff
@@ -324,5 +324,31 @@ public static String formatBytes(final long bytes) {
         return String.format(Locale.getDefault(), "%.1f %sB", bytes / Math.pow(1024, exp), pre);
     }
 
+    public static List<CharSequence> truncateCommonSubdir(final List<CharSequence> directories) {
```

Comment:
Do we have any guarantees on the length of `directories`? With length 0 we get an OutOfBoundsException and with length 1 the while loop will never terminate.